app/controllers/recipe_ingredients_controller.rb

```diff
@@ -0,0 +1,10 @@
+class RecipeIngredientsController < ApplicationController
+  def update
+    redirect = Recipe.find(params[:id])
+    recipe = Recipe.create!(name: params[:name], cost: params[:cost])
+    RecipeIngredient.create!(recipe_id: redirect.id, ingredient_id: recipe.id)
```

Comment:
This is the line we're looking for, except you're passing in a `recipe.id` (from line 4) into the `ingredient_id` key; as such this isn't accomplishing what hte user story is looking for, but you are close! 

You'll need a 2nd `.find` method, this time finding an Ingredient by its id. 

With stories like this, it's super important to keep track of any and all Ids you're sending into the controller.